libs/neon-shmem/Cargo.toml

```diff
@@ -8,6 +8,13 @@ license.workspace = true
 thiserror.workspace = true
 nix.workspace=true
 workspace_hack = { version = "0.1", path = "../../workspace_hack" }
+libc.workspace = true
+lock_api.workspace = true
+rustc-hash.workspace = true
 
 [target.'cfg(target_os = "macos")'.dependencies]
 tempfile = "3.14.0"
+
+[dev-dependencies]
+rand = "0.9"
```

Comment:
@erikgrinaker I didn't use the workspace crate for this part since Neon's on rand 0.8 which has a pretty significantly different API - I can port over the code though if it's important that we have everything in the workspace.